### PR TITLE
Remove hierarchy-related sidebar links

### PR DESCRIPTION
**Pull request by janfaracik**
Small PR to remove hierarchal sidebar links from every page that I could find them on. For context, the links removed are links such as 'Back to Dashboard' and 'Up'. We want to cut down on the amount of items in the sidebar and all these links do is duplicate breadcrumb items.

**Before**
image

**After**
image

**⚠️ Updated 16:30 25/07/22**

One-column layouts now appear centered + have their max-width increased, which I think helps make the layout easier to use + brings back the familiarity of the form not being on the far left of the screen. 

image

Let me know what y'all think 👍 

### Proposed changelog entries

* Remove hierarchal sidebar links

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6907"></a>

### CODE REVIEW

File: core/src/main/java/hudson/model/Computer.java
Comment (janfaracik) [inline core/src/main/java/hudson/model/Computer.java:776]: Nodes weren't linking correctly to the new Manage Jenkins parent url, this fixes that.

File: core/src/main/resources/hudson/logging/LogRecorderManager/all.jelly
Comment (janfaracik) [inline core/src/main/resources/hudson/logging/LogRecorderManager/all.jelly:33]: Small change just so the titles are consistent, happy to remove if reviewers want.

File: core/src/main/resources/hudson/logging/LogRecorderManager/sidepanel.jelly
Comment (janfaracik) [inline core/src/main/resources/hudson/logging/LogRecorderManager/sidepanel.jelly:36]: The link to All Log Messages is already in the table, so this is unnecessary.

File: core/src/main/java/hudson/model/Computer.java
Comment (daniel-beck) [reply core/src/main/java/hudson/model/Computer.java:776]: While non-admin users can access `/manage/whatever` (e.g. `/manage/cli/`, `/manage/computer/`), this will add a breadcrumb they are not entitled to access directly (`/manage/` performs a permission check).

File: core/src/main/resources/jenkins/tools/GlobalToolConfiguration/index.groovy
Comment (daniel-beck) [inline core/src/main/resources/jenkins/tools/GlobalToolConfiguration/index.groovy:16]: Time to apply the new config form to this page, so the sidepanel doesn't look so empty 😁

File: core/src/main/resources/hudson/model/UpdateCenter/sidepanel.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/model/UpdateCenter/sidepanel.jelly:33]: For pages that retain a sidepanel with links elsewhere, would it make sense to have a "You are here" item, like "Status" on the job overview page?

File: core/src/main/java/hudson/model/Computer.java
Comment (janfaracik) [reply core/src/main/java/hudson/model/Computer.java]: Ah I see. I've updated the getter so it'll check if the user has permission and if so return the Manage link. Let me know if there's any issues with that approach 👍

File: core/src/main/resources/hudson/model/UpdateCenter/sidepanel.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/model/UpdateCenter/sidepanel.jelly:33]: Good idea 👍 I'll look into that

**Review by timja — COMMENTED** [review]
These pages look weird and I think should be made one column:
- Global security
- Global Tools

This page looks a bit off but that's probably more that it either shouldn't exist or it should be a child page of the plugin manager I think:
- Update center

**Comment by daniel-beck** [comment]
> Global security
> Global Tools

They'll probably have the new form layout soon, no?

**Comment by timja** [comment]
> > Global security
> > Global Tools
> 
> They'll probably have the new form layout soon, no?

Possibly but for now they seem very unbalanced to me

File: core/src/main/resources/hudson/model/UpdateCenter/sidepanel.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/model/UpdateCenter/sidepanel.jelly]: Done 👍 Can be updated further once #6783 is in

image

**Comment by timja** [comment]
one-column layouts also seem off =/.

Plugin manager (which has stayed as 2 column currently) at least should be easy to fix by the existing PR that's open

File: core/src/main/resources/hudson/model/AbstractBuild/tasks.jelly
Comment (uhafner) [inline core/src/main/resources/hudson/model/AbstractBuild/tasks.jelly:30]: Shouldn't the status be removed as well?

File: core/src/main/resources/hudson/model/AbstractProject/sidepanel.jelly
Comment (uhafner) [inline core/src/main/resources/hudson/model/AbstractProject/sidepanel.jelly:40]: Shouldn't the status be removed as well?

**Review by uhafner — APPROVED** [review]
Maybe you should mention in the PR header that this change only affects Freestyle jobs 😜 I installed the PR several times until I noticed that there is duplicated code for Pipelines in the workflow plugins that obviously is not affected by your change.

**Comment by daniel-beck** [comment]
> Maybe you should mention in the PR header that this change only affects Freestyle jobs 😜 I installed the PR several times until I noticed that there is duplicated code for Pipelines in the workflow plugins that obviously is not affected by your change.

A recurring and not at all irrelevant problem for many of these PRs, without clear documentation and/or downstream PRs adapting plugins, Jenkins just becomes more inconsistent, and the benefits of such changes won't be seen by many users.

**Comment by timja** [comment]
> A recurring and not at all irrelevant problem for many of these PRs, without clear documentation and/or downstream PRs adapting plugins, Jenkins just becomes more inconsistent, and the benefits of such changes won't be seen by many users.

Can you file a downstream PR please @janfaracik?

File: core/src/main/resources/hudson/model/AbstractBuild/tasks.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/model/AbstractBuild/tasks.jelly:30]: The 'Status' link refers to the default child page for the project, and much like the updated Plugin Manager in #6783 all children pages should be there for easy navigation between them. I've updated the Design Library docs for this 👍

Comment (uhafner) [reply core/src/main/resources/hudson/model/AbstractBuild/tasks.jelly:30]: I see. Then it makes sense to let it unchanged.

Comment (jglick) [inline core/src/main/resources/hudson/model/AbstractBuild/tasks.jelly:30]: Could then deprecate https://github.com/jenkinsci/jenkins/blob/082634cf1e4ef91cac8c82815e52046083722e36/core/src/main/java/hudson/model/AbstractBuild.java#L263-L277 I think. (Still used by some plugins, so cannot be deleted yet.)

File: war/src/main/less/base/layout-commons.less
Comment (timja) [inline war/src/main/less/base/layout-commons.less:72]: something is off with the configure security page, (at least on small screens, i.g. 13" mac pro chrome).

the inputs use the full width to the right, whereas the others are balanced

image

Comment (janfaracik) [reply war/src/main/less/base/layout-commons.less:72]: Does it do that if authorisation isn't set to matrix-based security?

Comment (timja) [reply war/src/main/less/base/layout-commons.less:72]: it works right if I change it to something else, yes nice find, some css need fixing there?

Comment (janfaracik) [reply war/src/main/less/base/layout-commons.less:72]: Should be fixed now 👍 🤞

File: core/src/main/resources/hudson/model/AbstractBuild/tasks.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/model/AbstractBuild/tasks.jelly:30]: Done, cheers! 👍

**Comment by timja** [comment]
ATH results need checking:
https://github.com/jenkinsci/acceptance-test-harness/pull/869

File: core/src/main/java/hudson/model/Computer.java
Comment (daniel-beck) [reply core/src/main/java/hudson/model/Computer.java]: I cannot think of any off hand, but it feels kinda icky. At least it's not as bad as what we do for jobs in views 😁

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
checkstyle issue and ATH to sort

File: core/src/main/java/hudson/model/AbstractBuild.java
Comment (jglick) [inline core/src/main/java/hudson/model/AbstractBuild.java:91]:
```suggestion
```

**Comment by timja** [comment]
ATH is sorted in https://github.com/jenkinsci/acceptance-test-harness/pull/869

----

/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by daniel-beck** [comment]
Any plans to adapt the most popular plugins to this new "style guide"? Is there developer documentation for this change?

**Comment by janfaracik** [comment]
> Any plans to adapt the most popular plugins to this new "style guide"? Is there developer documentation for this change?

Will do 👍 There's some guidance in draft here https://github.com/jenkinsci/design-library-plugin/pull/90

**Comment by basil** [comment]
Causes [JENKINS-69370](https://issues.jenkins.io/browse/JENKINS-69370) (CC @janfaracik). This regression affects the connection of inbound agents. In the absence of a verified fix by the end of the week, I plan to revert this PR prior to next week's release of 2.365.

**Pull request merged**
